src/auth/User.cc

```diff
@@ -125,104 +79,43 @@ Auth::User::~User()
     /* free cached acl results */
     aclCacheMatchFlush(&proxy_match_cache);
 
-    /* free seen ip address's */
-    clearIp();
-
     if (username_)
         xfree((char*)username_);
 
     /* prevent accidental reuse */
     auth_type = Auth::AUTH_UNKNOWN;
 }
 
+/// generate the cache key for an ipList entry
+static const SBuf
+BuildIpKey(const Ip::Address &ip)
+{
+    SBuf key;
+    auto *buf = key.rawAppendStart(MAX_IPSTRLEN);
+    const auto len = ip.toHostStr(buf, MAX_IPSTRLEN);
+    key.rawAppendFinish(buf, len);
+    return key;
+}
+
 void
 Auth::User::clearIp()
 {
-    AuthUserIP *ipdata, *tempnode;
-
-    ipdata = (AuthUserIP *) ip_list.head;
-
-    while (ipdata) {
-        tempnode = (AuthUserIP *) ipdata->node.next;
-        /* walk the ip list */
-        dlinkDelete(&ipdata->node, &ip_list);
-        delete ipdata;
-        /* catch incipient underflow */
-        assert(ipcount);
-        -- ipcount;
-        ipdata = tempnode;
-    }
-
-    /* integrity check */
-    assert(ipcount == 0);
+    const auto savedLimit = ipList.memLimit();
+    ipList.setMemLimit(0);
+    ipList.setMemLimit(savedLimit);
```

Comment:
Please add an efficient and trivial clear() method and use it here.

```suggestion
    ipList.clear();
```